### ISSUE TITLE
solution option interferes with echo option

### ISSUE THREAD

**Issue by nickreich** [koncina/unilur#8]
When I have a setup chunk that sets echo=FALSE, this seems to override the solution=TRUE option, unless I define echo=TRUE in each solution chunk. In the example test.txt file below (actually an .Rmd file but I could only upload txt), the first solution does not show. The second does. If nothing else, maybe just let folks know that this is an issue..

[test.txt](https://github.com/koncina/unilur/files/1458176/test.txt)

![image](https://user-images.githubusercontent.com/1280767/32614454-4bf791fa-c53b-11e7-93c9-9b112d7d990f.png)

**Comment by nickreich** [comment]
Otherwise, I want to say thanks for an awesome tool! This is just what I was looking for to use for my exams! And this minor bug (with an existing work-around) won't prevent me from using it. Thanks!

**Comment by koncina** [comment]
Thank you for your feedback.

I don't think that having to explicitly set `echo = TRUE` after using `knitr::opts_chunk$set(echo = FALSE)` is an issue.
This might sound counterintuitive for `asis` or `block` chunks (maybe in particular in the `solution` context) but as the base formats `html_document()` and `pdf_document()` also behave like this (they will not show up with `echo = FALSE`) changing this would turn the setting inconsistent IMHO.
Being able to set `echo` and/or `eval` even for a solution chunk allows some fine-tuning. You can for example show a plot as a solution but still hide the source code generating it like here:

~~~
```{r, solution = TRUE, echo = FALSE}
plot(cars)
```
~~~

![image](https://user-images.githubusercontent.com/6618930/32652881-2d406ebe-c606-11e7-96a9-acfa87886e78.png)


You nevertheless pointed out an issue that I will fix: If the solution is empty (as in your example or if you're using a _R_ chunk with the settings `echo = FALSE` and `eval = FALSE`), it makes no sense to draw the green solution box.

**Comment by nickreich** [comment]
I agree that it isn't a bug, but it took me about 30 minutes of debugging to figure out that I needed to use echo=TRUE in a chunk after setting it as globally false in a setup chunk. Maybe the "issue" is that it would be useful to add this use-case (which I suspect is something others may run into) as part of the documentation.
